docs/proposals/policy-feg-api-cleanup/qos_enforcement.md

```diff
@@ -0,0 +1,398 @@
+The final state is summarized here:
+
+```
+policy_qos_profile:
+  id: string
+  class_id: enum[0,255]
+  max_req_bw_dl: uint32
+  max_req_bw_ul: uint32
+  # Guaranteed bit rate
+  # Not enforced by Magma yet
+  gbr:
+    gbr_ul: uint32
+    gbr_dl: uint32
+  # Allocation and retention priority
+  # Not enforced by Magma yet
+  arp:
+    arp_priority_level: uint32
+    arp_preemption_capability: bool
+    arp_preemption_vulnerability: bool
+```
+
+This is an addition, as there is an existing `qos_profile` definition, which
+will continue to be used only for the `apn_configuration` entity.
+
+This `policy_qos_profile` will be marshalled to the `FlowQos` protobuf
+message.
+
+```
+flow_qos:
+  ...
+```
+
+The `policy_qos_profile` will replace the `flow_qos` swagger entity
+in funtctionality.
+`flow_qos` should be marked as deprecated and should be discouraged from further
+use in the API.
+
+```
+apn_configuration:
+  ...
+  # Unchanged
+  ...
+```
+
+As before, a single QoS profile will be associated to an APN object at creation.
+This will define the QoS for the default bearer, in combination with the `ambr`
+field, and is unchanged.
+
+```
+apn_resource:
+  id: string
+  # Identifies that the SGi interface is set up for the specified APN
+  apn_name: string
+  # Specify either VLAN or the interface name for SGi
+  # VLAN to use when communicating over SGi interface
+  vlan_id: string
+```
+
+The `apn_resource` defines how the APN is serviced by each access gateway.
+Options are provided for IP allocation, and the uplink. It is possible that
+two APNs are serviced by separate uplink interfaces on the same access gateway.
+
+```
+mutable_lte_gateway:
+  id: string
+  name: string
+  description: string
+  ...
+  apn_resources: array[string]
+  ...
+```
+
+We are adding the `apn_resources` field to `lte_gateway` and
+`mutable_lte_gateway` and they are specified by ID.
+
+```
+network_epc_configs:
+  ...
+  hss_relay_enabled: bool
+  gx_gy_relay_enabled: bool
+  ...
+```
+
+The `relay_enabled` flag will be split to enable federation under different
+circumstances, with finer control. The combination of `hss_relay_enabled` and
+`gx_gy_relay_enabled` together have the same functionality as the current
+`relay_enabled`.
+
+```
+subscriber:
+  id: string
+  name: string
+  ...
+  active_base_names: array[base_name]
+  # Policy IDs listed in active_policies define which policies will be installed
+  # for every APN of the subscriber
+  active_policies: array[string]
+  # Keyed by APN IDs, and values list out the policy IDs
+  # Keys present define the APNs that the subscriber has access to
+  # policy IDs define what policies will be installed for the subscriber for 
+  # each APN
+  active_policies_by_apn: dict[string, array[string]]
+  ...
+```
+
+To now specify APN-specific policies, the field `active_policies_by_apn` will
+be added. If an APN key is not present in `active_policies_by_apn`, then the
+subscriber does not have access to that APN.
```

Comment:
I missed this comment. So are we removing active_apns? How do we then enable subscriber access with default APN policy? Do we have a special name for it?

Reply:
Note for future: as resolved offline, this line should actually be "If an APN key is not present in `active_apns`, then the subscriber does not have access to that APN